app/src/main/cpp/skyline/applet/swkbd/software_keyboard_applet.cpp

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2022 Skyline Team and Contributors (https://github.com/skyline-emu/)
+// Copyright © 2019-2022 Ryujinx Team and Contributors
+
+#include <codecvt>
+#include <services/am/storage/ObjIStorage.h>
+#include <common/settings.h>
+#include "software_keyboard_applet.h"
+#include <jvm.h>
+
+class Utf8Utf16Converter : public std::codecvt<char16_t, char8_t, std::mbstate_t> {
+  public:
+    ~Utf8Utf16Converter() override = default;
+};
+
+namespace skyline::applet::swkbd {
+    InteractiveOutputResult::InteractiveOutputResult(std::u16string_view text, bool useUtf8Storage) : size(sizeof(InteractiveOutputResult)), chars() {
+        if (useUtf8Storage) {
```

Comment:
Make this a helper as it's just duplicated code with normal output result, also use text.end,().base() 
For chars split out casted version into a local variable and reference that, also use size_bytes and comment that you're null terminating jnput